codec-http2/src/main/java/io/netty/handler/codec/http2/Http2HeadersEncoder.java

```diff
@@ -22,7 +22,16 @@
  * Encodes {@link Http2Headers} into HPACK-encoded headers blocks.
  */
 @UnstableApi
-public interface Http2HeadersEncoder {
+public interface Http2HeadersEncoder extends AutoCloseable {
+
+    /**
+     * Close the encoder and release all its associated data.
+     *
+     * @throws IllegalStateException If this encoder has already been closed.
+     */
+    @Override
+    void close();
+
```

Comment:
This is a breaking change ... Let's just implement this interface directly and add an instanceof check

Reply:
Oh, sorry. 

I thought it would be OK since these types are marked by `@UnstableApi`